src/main/java/org/apache/commons/lang3/LocaleUtils.java

```diff
@@ -324,7 +327,7 @@ public static Locale toLocale(final String str) {
             throw new IllegalArgumentException("Invalid locale format: " + str);
         }
         final char ch0 = str.charAt(0);
-        if (ch0 == '_') {
+        if (ch0 == '_' || ch0 == '-') {
```

Comment:
We could use "UNDERSCORE" or "DASH" here or introduce a helper method, e.g. "isDelimiter"? What do you think?

Reply:
Added the constant in 11a6d1d52. I'd prefer not to add the helper method to avoid changing too much of the structure as part of this fairly minor overall change.